Cards/CardBase.gd

```diff
@@ -42,8 +45,46 @@ func on_card_play(caster: Entity, base_target: Entity) -> void:
 	if caster is Player:
 		PlayerManager.player.get_energy_component().use_energy(self)	
 	
-	for effect_data: EffectData in card_effects_data:
-		var list_targets: Array[Entity] = effect_data.targeting_function.generate_target_list(base_target)
-		for current_target in list_targets:
-			effect_data.apply_effect_data(caster, current_target)
-	CardManager.on_card_action_finished.emit(self)
+	if _card_effects_queue.size() > 0:
+		push_error("Tried to play more card animations while some are already playing!")
+		return
+	
+	_card_effects_queue = card_effects_data.duplicate()
+	_handle_effects_queue(caster, base_target)
+
+
+func _handle_effects_queue(caster: Entity, base_target: Entity) -> void:
+	var card_effect: EffectData = _card_effects_queue[0]
+	var animation_data: CastAnimationData = card_effect.animation_data
+	var list_targets: Array[Entity] = card_effect.targeting_function.generate_target_list(base_target)
+	
+	var can_use_animation: bool = animation_data != null and\
+		animation_data.cast_animation_scene != null and\
+		animation_data.cast_position != null and\
```

Comment:
I think we can encapsulate these animation data checks within the CastAnimationData itself, makes it a bit easier if we're using this in other places so we're not copypasta-ing the logic.

Reply:
agreed

Reply:
Good call, i'll move it